### PR TITLE
Website

### PR DESCRIPTION
**Pull request by cb372**
I felt that the README was getting pretty unwieldy, and I received a few comments from colleagues complaining that the examples were not very beginner-friendly, so I decided to take a stab at creating a documentation site for circe.

I've also expanded the documentation with a few (hopefully) beginner-friendly examples.

I'm not sure how controversial this is, but I also decided to make optics a first-class citizen in the documentation, as I find them to be (a) very useful, and (b) quite confusing for beginners without any examples to follow.

Still a work in progress, but I thought I'd garner feedback early.

What I've done so far:

* Shamelessly steal the following from Cats
    * Stylesheets/visual design
    * Jekyll site structure
    * sbt code to wire everything up
* Chop up README.md and CONTRIBUTING.md into a few pages
* Add some more examples and tut-ify existing ones

TODO:

- [x] Does circe have a logo?!
- [x] Write the rest of the Optics page
- [x] Explain the differences between the types of cursor
- [x] Check the README for anything I missed when I was porting the content across
- [ ] Dedicated page for Scala.js? I've barely used Scala.js, so I'm not really qualified to write any documentation about it.
- [ ] Some fresh benchmark data would be nice, maybe once 0.5.0 is released?

Here's what the site looks like so far:
https://cb372.github.io/circe/

### CODE REVIEW

File: docs/src/site/index.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+title:  "Home"
+section: "home"
+---
+
+[![Build status](https://img.shields.io/travis/travisbrown/circe/master.svg)](https://travis-ci.org/travisbrown/circe)
+[![Coverage status](https://img.shields.io/codecov/c/github/travisbrown/circe/master.svg)](https://codecov.io/github/travisbrown/circe)
+[![Gitter](https://img.shields.io/badge/gitter-join%20chat-green.svg)](https://gitter.im/travisbrown/circe)
+[![Maven Central](https://img.shields.io/maven-central/v/io.circe/circe-core_2.11.svg)](https://maven-badges.herokuapp.com/maven-central/io.circe/circe-core_2.11)
+
+circe (pronounced SUR-see, or KEER-kee in classical Greek, or CHEER-chay in Ecclesiastical Latin) is
```
Comment (som-snytt) [inline docs/src/site/index.md]: Maybe `KEER-keh` for ancient Greek.

**Comment by travisbrown** [comment]
Thanks again, @cb372, this is great! And yes, we do have a logo, thanks to @47deg: https://meta.plasm.us/tmp/circe.svg

We haven't really launched it yet, but the new site seems like a good reason. :)

**Comment by cb372** [comment]
@travisbrown Cool! I don't really understand why it's a bird though. Didn't Circe turn people into pigs?

This is now ready for review. I've updated https://cb372.github.io/circe/ with the latest changes.

I know documentation is a very personal thing, so feel free to nitpick as much as you like. Anything from adding an Oxford comma to rewriting whole sections. I can also squash commits if you want.

**Comment by cb372** [comment]
@travisbrown I've rebased to fix a merge conflict, and added some more docs about encoding/decoding as suggested in #350 and #213.

Let me know if there are any other blockers to merging this.

By the way I noticed that Monocle uses Travis CI to [push the site automatically](https://github.com/julien-truffaut/Monocle/blob/88957e406664530f5ec9072228923a6442a1bf67/.travis.yml#L37-L38). That seems like quite a good idea.

**Comment by cb372** [comment]
@travisbrown I've rebased again and updated for 0.5.1.

Is there any chance of getting this merged soon? Rebasing is quite painful, as I have to manually port any README changes over to the site every time.

**Comment by travisbrown** [comment]
Thanks again @cb372! I'll aim to get this merged today—I'll have some time now that the releases are out.

**Comment by travisbrown** [comment]
@cb372 I've been thinking about this in more detail this afternoon, and the one change I want to propose is moving the more meta stuff (specifically the license, community, contributors sections) back to the README / plain old GitHub markdown pages.

There are a couple of motivations for this—the CONTRIBUTING.md file gets [special treatment](https://github.com/blog/1184-contributing-guidelines) in GitHub, and having the adopters section in the README makes the barrier to people adding their organizations a little lower.

This split (compiled library user guide site vs. GitHub markdown pages for information _about_ the project as a project) also follows the pattern of Finagle and other projects I've worked on. What do you think? I'm happy to open a pull request against your branch this afternoon if this sounds reasonable.

**Comment by cb372** [comment]
That sounds fine to me. Go ahead, if you're happy to do the work.

Just be aware that I amalgamated all the markdown references from README.md and CONTRIBUTING.md into a single file, so unpicking them again might be a bit fiddly. Sorry about that.

**Pull request merged**